nyx_client/nyx_client/client.py

```diff
@@ -937,14 +937,19 @@ def delete_circle(self, circle: Circle):
         """
         self.delete_circle_by_name(circle.name)
 
-    def get_connections(self) -> list[Connection]:
+    def get_connections(self, allow_upload: bool | None = None) -> list[Connection]:
         """Lists all connections.
 
+        Args:
+            allow_upload: Filter the connections by allow_upload, defaults to show all.
```

Comment:
I'd say the only thing is it might be clear to uses that there are three options here:
1. `None` - all
2. `allow_upload=True` just with upload enabled (write)
3. `allow_upload=False` just with upload disabled (read-only)

But maybe it doesn't matter.